### PR TITLE
feat(gpg): add completion spec

### PR DESCRIPTION
**Pull request by bolshoytoster**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Adds specs for `gpg`.
**What is the current behavior? (You can also link to an open issue here)**
`gpg` does not have completions #890.
**What is the new behavior (if this is a feature change)?**
`gpg` does have completions
**Additional info:**

Adds gpg completions (taken straight from the man page).
Adds two generators, both use `gpg --version` to get supported algorithms

### CODE REVIEW

File: src/gpg.ts [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+      description: "Encrypt for user id name",
+      args: { name: "name" },
+    },
+    {
+      name: "--default-recipient",
+      description: "Use name as default recipient",
+      args: { name: "name" },
+    },
+    { name: "--default-recipient-self", description: "Use the default key as default recipient" },
+    { name: "--no-default-recipient", description: "Reset --default-recipient and --default-recipient-self" },
```
Comment (fedeci) [inline src/gpg.ts:66]: This option is exclusive on its counterpart and vice versa:
```suggestion
    { name: "--no-default-recipient", exclusiveOn: ["--default-recipient", "--default-recipient-self"], description: "Reset --default-recipient and --default-recipient-self" },
```

File: src/gpg.ts [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+      name: ["-r", "--recipient"],
+      description: "Encrypt for user id name",
+      args: { name: "name" },
+    },
+    {
+      name: "--default-recipient",
+      description: "Use name as default recipient",
+      args: { name: "name" },
+    },
+    { name: "--default-recipient-self", description: "Use the default key as default recipient" },
```
Comment (fedeci) [inline src/gpg.ts:65]:
```suggestion
    { name: "--default-recipient-self", exclusiveOn: ["--no-default-recipient"], description: "Use the default key as default recipient" },
```

File: src/gpg.ts [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+      description: "Use name as default user ID for signatures",
+      args: { name: "name" },
+    },
+    {
+      name: ["-r", "--recipient"],
+      description: "Encrypt for user id name",
+      args: { name: "name" },
+    },
+    {
+      name: "--default-recipient",
```
Comment (fedeci) [inline src/gpg.ts:61]:
```suggestion
      name: "--default-recipient",
      exclusiveOn: ["--no-default-recipient"],
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Generally it looks good, can you add all the missing `exclusiveOn`, please?

File: package.json [hunk 4]
```diff
@@ -97,6 +97,7 @@
     "@types/glob": "^7.2.0",
     "fs": "^0.0.1-security",
     "glob": "^7.2.0",
+    "optionator": "^0.9.1",
```
Comment (fedeci) [inline package.json:100]:
```suggestion
```

**Review by fedeci — CHANGES_REQUESTED** [review]

**Comment by fedeci** [comment]
> Generally it looks good, can you add all the missing `exclusiveOn`, please?

There are many more options that has a `--no-` counterpart and those are still missing the `exclusiveOn`, the three ones I suggested were just examples😄. Don't forget to run `npm run lint:fix` before pushing, so the CI is happy :)

File: package.json (hunk 4 above)
Comment (bolshoytoster) [reply package.json:100]: @fedeci sorry, I accidentally did that.

**Review by fedeci — APPROVED** [review]
Thanks! `npm run lint:fix` and we are done

**Comment by bolshoytoster** [comment]
@fedeci
> Thanks! `npm run lint:fix` and we are done

I ran that and there was no change.

**Comment by fedeci** [comment]
Probably master is failing, I am merging this!

**Pull request merged**